File: C++/factorial.cpp
#include <iostream>
using namespace std;

class Factorial
{
private:
    int num;
    int factorial = 1;

public:
    void calculateFactorial();
    void show();
};

void
Factorial::calculateFactorial()
{
    printf("Enter a number : \n" );
    scanf("%d",&num);

    if (num == 0 || num == 1)
    {
        factorial = 1;
    }
    else
    {
        while (num > 1)
        {
            factorial = factorial * num;
            num--;
        }
    }
}

void Factorial::show()
{
    printf("Factorial : %d\n", factorial);
}

int main()
{
    Factorial factorial;
    factorial.calculateFactorial();
    factorial.show();
}

File: Python/factorial.py
#factorial program
n=int(input('Enter a number : '))
fact=1
if n>0:
    for i in range(1,n+1):
        fact=fact*i
    print('Factorial value : ' + str(fact))
    
elif n<0:
    print('Invalid number..Enter a positive number!!')
    
else:
    print('Factorial value : ' + str(fact))
